### PR TITLE
refactor: sdk getting started & example update

### PR DESCRIPTION
**Pull request by Dudleyneedham**
## fixes https://github.com/KILTprotocol/ticket/issues/1589

Updating the getting started and example guides to include the did refactor
Removing actors from the example guide.
Adding package json to the docs

## How to test:

Run through the getting started and example checking against a running chain.

## Checklist:

- [ ] I have verified that the code works
- [ ] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [ ] I have [left the code in a better state](https://deviq.com/principles/boy-scout-rule)
- [ ] I have documented the changes (where applicable)

### CODE REVIEW

File: docs/example.ts [hunk 1]
```diff
@@ -27,10 +42,20 @@ async function setup(): Promise<{
 
   // To get an attestation, we need an Attester
   // we can generate a new keypair:
-  // const attester = await Kilt.AttesterIdentity.buildFromMnemonic("...")
+
+  const keyring = new Kilt.Utils.Keyring({
+    // KILT has registered the ss58 prefix 38
+    ss58Format: 38,
+    type: 'ed25519',
```
Comment (Diiaablo95) [inline docs/example.ts:49]: Should we use `sr25519`?

File: docs/example.ts [hunk 2]
```diff
@@ -27,10 +42,20 @@ async function setup(): Promise<{
...
+  const attesterMnemonic =
+    'receive clutch item involve chaos clutch furnace arrest claw isolate okay together'
+
   // or we just use unsafe precalculated keys (just for demo purposes!):
-  const attester = Kilt.Identity.buildFromMnemonic(
-    'receive clutch item involve chaos clutch furnace arrest claw isolate okay together',
+  const attester = keyring.addFromMnemonic(
+    attesterMnemonic,
     // using ed25519 bc this identity has test coins from the start on the development chain spec
     { signingKeyPairType: 'ed25519' }
```
Comment (Diiaablo95) [inline docs/example.ts:60]: Same here?

File: docs/example.ts [hunk 3]
```diff
@@ -63,20 +101,27 @@ async function setup(): Promise<{
...
+    owner: attesterOnChainDid.did,
   }
 
   // Build the CType object
   const ctype = new Kilt.CType(rawCtype)
-
   // Store ctype on blockchain
   // signAndSubmitTx can be passed SubscriptionPromise.Options, to control resolve and reject criteria, set tip value, or activate re-sign-re-send capabilities.
   // ! This costs tokens !
   // Also note, that the completely same ctype can only be stored once on the blockchain.
```
Comment (Diiaablo95) [inline docs/example.ts:112]: I would rephrase this to just say "the same CType". No need to specify completely, in my opinion.

File: docs/example.ts [hunk 4]
```diff
@@ -86,191 +131,329 @@ async function setup(): Promise<{
 
   // ------------------------- Claimer  ----------------------------------------
   // How to generate an Identity
-  // const mnemonic = Kilt.Identity.generateMnemonic()
-  const claimer = Kilt.Identity.buildFromMnemonic(
+  // const mnemonic = Kilt.Utils.UUID.generate()
```
Comment (Diiaablo95) [inline docs/example.ts:134]: Can we remove these comments?

File: docs/example.ts [hunk 5]
```diff
@@ -86,191 +131,329 @@ async function setup(): Promise<{
...
+  const claimerMnemonic =
     'wish rather clinic rather connect culture frown like quote effort cart faculty'
-  )
-  // const address = claimer.address
+
+  // Create a light DID from the generated authentication key.
```
Comment (Diiaablo95) [inline docs/example.ts:138]:
```suggest
// Generate authentication and encryption keys and then derive a light DID from them.
```

Or something like that.

File: docs/example.ts [hunk 6]
```diff
@@ -86,191 +131,329 @@ async function setup(): Promise<{
...
+  const claimerSigningKeypair = await keystore.generateKeypair({
+    alg: Kilt.Did.SigningAlgorithms.Ed25519,
+    seed: claimerMnemonic,
+  })
+  const claimerEncryptionKeypair = await keystore.generateKeypair({
+    alg: Kilt.Did.EncryptionAlgorithms.NaclBox,
+    seed: claimerMnemonic,
+  })
+  // Create a light DID from the generated authentication key.
```
Comment (Diiaablo95) [inline docs/example.ts:147]: Same here...

File: docs/example.ts [hunk 7]
```diff
@@ -86,191 +131,329 @@ async function setup(): Promise<{
...
+    },
+    encryptionKey: {
+      publicKey: claimerEncryptionKeypair.publicKey,
+      type: Kilt.Did.DemoKeystore.getKeypairTypeForAlg(
+        claimerEncryptionKeypair.alg
+      ),
+    },
+  })
 
   // At this point the generated Identity has no tokens.
```
Comment (Diiaablo95) [inline docs/example.ts:163]: Identity?

File: docs/example.ts [hunk 8]
```diff
@@ -86,191 +131,329 @@ async function setup(): Promise<{
...
   // If you want to interact with the blockchain, you will have to get some.
   // Contact faucet@kilt.io and provide the address of the identity
```
Comment (Diiaablo95) [inline docs/example.ts:165]:
```suggest
  // Contact faucet@kilt.io and provide the address of the identity.
```

File: docs/example.ts [hunk 9]
```diff
@@ -86,191 +131,329 @@ async function setup(): Promise<{
...
+  const claimerRequestMessage = new Kilt.Message(
+    {
+      type: Kilt.Message.BodyType.REQUEST_ATTESTATION_FOR_CLAIM,
+      content: { requestForAttestation },
+    },
+    claimerLightDid.did,
+    attesterOnChainDid.did
   )
 
+  const claimerKeyAgreement = claimerLightDid.getKeys(
```
Comment (Diiaablo95) [inline docs/example.ts:222]:
```suggest
  const claimerKeyAgreementKey = claimerLightDid.getKeys(
```

File: docs/example.ts [hunk 10]
```diff
@@ -86,191 +131,329 @@ async function setup(): Promise<{
...
+    KeyRelationship.keyAgreement
+  )[0] as IDidKeyDetails<string>
+  const attesterKeyAgreement = attesterOnChainDid.getKeys(
```
Comment (Diiaablo95) [inline docs/example.ts:225]:
```suggest
  const attesterKeyAgreementKey = attesterOnChainDid.getKeys(
```

File: docs/example.ts [hunk 11]
```diff
@@ -86,191 +131,329 @@ async function setup(): Promise<{
...
+  )
+  // Attester can check the data and verify the data has not been tampered with
+  if (!claimersRequest.verifyData()) {
+    console.log('data is false')
+  }
 
-  // And make sure, that the sender is the owner of the identity
-  Kilt.Message.ensureOwnerIsSender(reqAttestationDec)
+  // Attester can check if the signature of the claimer matches the request for attestation object
+  claimersRequest.verifySignature()
```
Comment (Diiaablo95) [inline docs/example.ts:257]:
```suggest
  await claimersRequest.verifySignature()
```

File: docs/example.ts [hunk 12]
```diff
@@ -86,191 +131,329 @@ async function setup(): Promise<{
...
+
+  const verifierSigningKeypair = await keystore.generateKeypair({
+    alg: Kilt.Did.SigningAlgorithms.Ed25519,
+    seed: verifierMnemonic,
+  })
+  const verifierEncryptionKeypair = await keystore.generateKeypair({
+    alg: Kilt.Did.EncryptionAlgorithms.NaclBox,
+    seed: verifierMnemonic,
+  })
+  // Create a light DID from the generated authentication key.
```
Comment (Diiaablo95) [inline docs/example.ts:337]: And encryption keys as well?

File: docs/example.ts [hunk 13]
```diff
@@ -86,191 +131,329 @@ async function setup(): Promise<{
...
+    },
+    encryptionKey: {
+      publicKey: verifierEncryptionKeypair.publicKey,
+      type: Kilt.Did.DemoKeystore.getKeypairTypeForAlg(
+        verifierEncryptionKeypair.alg
+      ),
+    },
+  })
+
+  const claimerKeyAgreement = claimerLightDid.getKeys(
```
Comment (Diiaablo95) [inline docs/example.ts:353]:
```suggest
  const claimerKeyAgreementKey = claimerLightDid.getKeys(
```

File: docs/example.ts [hunk 14]
```diff
@@ -86,191 +131,329 @@ async function setup(): Promise<{
...
+    KeyRelationship.keyAgreement
+  )[0] as IDidKeyDetails<string>
+  const verifierKeyAgreement = verifierLightDid.getKeys(
```
Comment (Diiaablo95) [inline docs/example.ts:356]:
```suggest
  const verifierKeyAgreement = verifierLightDid.getKeys(
```

File: docs/example.ts [hunk 15]
```diff
@@ -86,191 +131,329 @@ async function setup(): Promise<{
...
+
+  const verifierSubmitClaimsMessageDec = await Kilt.Message.decrypt(
+    claimerSubmitClaimsMessageEnc,
+    keystore,
+    { senderDetails: claimerLightDid, receiverDetails: verifierLightDid }
+  )
+  const presentationMessage = (verifierSubmitClaimsMessageDec.body as ISubmitClaimsForCTypes)
+    .content
+
+  const verifierablePresentation = Kilt.AttestedClaim.fromAttestedClaim(
```
Comment (Diiaablo95) [inline docs/example.ts:422]:
```suggest
  const verifiablePresentation = Kilt.AttestedClaim.fromAttestedClaim(
```

File: docs/example.ts [hunk 16]
```diff
@@ -86,191 +131,329 @@ async function setup(): Promise<{
...
+    presentationMessage[0]
   )
-  console.log('Received claims: ', JSON.stringify(claims))
+  const verified = await verifierablePresentation.verify({ challenge })
```
Comment (Diiaablo95) [inline docs/example.ts:425]:
```suggest
  const verified = await verifiablePresentation.verify({ challenge })
```

File: docs/getting-started.md [hunk 17]
```diff
@@ -7,18 +7,22 @@ The next examples give you a simple skeleton on how to use the KILT SDK to creat
 
 - [1. How to install the SDK](#1-how-to-install-the-sdk)
   - [1.1. Prerequisites](#11-prerequisites)
+  - [1.2. Connect to a Chain](#11-connect-to-a-chain)
 - [2. How to generate an Identity](#2-how-to-generate-an-identity)
-- [3. How to build and store a Claim Type (CTYPE)](#3-how-to-build-and-store-a-claim-type-ctype)
+  - [2.1. Generate a Keystore](#21-generate-a-keystore)
+  - [2.2. Generate a light DID for the Claimer](#22-generate-a-light-did-for-the-claimer)
+  - [2.3. Generate a full DID for the Attester](#22-generate-a-full-did-for-the-attester)
+* [3. How to build and store a Claim Type (CTYPE)](#3-how-to-build-and-store-a-claim-type-ctype)
```
Comment (Diiaablo95) [inline docs/getting-started.md:15]: Should we just call it `Build and store...`? Not beginning with how, as it is not used anywhere else (if we also change at point 4 and 5).

File: docs/getting-started.md [hunk 18]
```diff
@@ -80,99 +84,285 @@ await Kilt.init({ address: YOUR_CHAIN_ADDRESS })
...
+Or by setting a default address in the configuration, connecting implicitly.
+
+```typescript
+Kilt.config({ address: YOUR_CHAIN_ADDRESS })
+```
+
+Note that calling (as described in [1.2](#initializing-the-kilt-sdk))
+
+```typescript
+Kilt.init({ address: YOUR_CHAIN_ADDRESS })
```
Comment (Diiaablo95) [inline docs/getting-started.md:105]: Should this be prefixed with `await`? And maybe also the previous calls?

File: docs/getting-started.md [hunk 19]
```diff
@@ -80,99 +84,285 @@ await Kilt.init({ address: YOUR_CHAIN_ADDRESS })
...
+```
+
+initializes the SDK _and_ sets the config, so it is related to the second approach.
+
+There are of KILT chains which you can use, each one having a different address:
+
+1. A local node: `ws://127.0.0.1:9944`
+2. The prod-net: `wss://peregrine.kilt.io`
```
Comment (Diiaablo95) [inline docs/getting-started.md:113]: Maybe we should rename this to something else?

File: docs/getting-started.md [hunk 20]
```diff
@@ -80,99 +84,285 @@ await Kilt.init({ address: YOUR_CHAIN_ADDRESS })
...
+3. The dev-net: `wss://kilt-peregrine-stg.kilt.io`
+4. The live-net: `wss://spiritnet.kilt.io`
+
+In case you go with option #1 or #2, you have to request test money **since storing a CTYPE on the chain requires tokens and a full did** as transaction fee and deposit.
```
Comment (Diiaablo95) [inline docs/getting-started.md:117]: Not option #3 as well?

File: docs/getting-started.md [hunk 21]
```diff
@@ -80,99 +84,285 @@ await Kilt.init({ address: YOUR_CHAIN_ADDRESS })
...
+However, **we recommend to start your local node** and use a mnemonic which already has tokens by using our docker image. Depending on which version of the SDK you are using, you might need to spin up the latest officially released version of the node (working with the latest release of the SDK) or the latest development version of the node (working with the latest development version of the SDK). Either version can be started with the following command by using one between the `latest` and `develop` tag.
```
Comment (Diiaablo95) [inline docs/getting-started.md:118]: I am not sure this is clear for users. It might be very confusing. Not saying we should remove this, but perhaps we could ask someone else's opinion on this?

File: docs/getting-started.md [hunk 22]
```diff
@@ -80,99 +84,285 @@ await Kilt.init({ address: YOUR_CHAIN_ADDRESS })
...
+
+**Option #4 is the live net work and not recommend to test on as it requires tokens with real value**
```
Comment (Diiaablo95) [inline docs/getting-started.md:120]:
```suggest
**Option #4 is the live network and not recommended to test on as it requires tokens with real value**
```

File: docs/getting-started.md [hunk 23]
```diff
@@ -80,99 +84,285 @@ await Kilt.init({ address: YOUR_CHAIN_ADDRESS })
...
+
+```
+docker run -p 9944:9944 kiltprotocol/peregrine:{latest,develop} --dev --ws-port 9944 --ws-external --rpc-external
```
Comment (Diiaablo95) [inline docs/getting-started.md:123]: `--tmp` as well?

File: docs/getting-started.md [hunk 24]
```diff
@@ -80,99 +84,285 @@ await Kilt.init({ address: YOUR_CHAIN_ADDRESS })
...
+const identity = keyring.addFromMnemonic(
+  'receive clutch item involve chaos clutch furnace arrest claw isolate okay together',
+  // depending on what setup either use ed25519 or sr25519 as key type because this is how the endowed identity is set up
+  { signingKeyPairType: 'ed25519' }
+)
+```
+
+## 2. How to generate an Identity and DID
+
+To generate an Identity first you have to generate a [BIP39 mnemonic](https://github.com/bitcoin/bips/blob/master/bip-0039.mediawiki) and then use it to create the on-chain Identity and DID:
```
Comment (Diiaablo95) [inline docs/getting-started.md:139]: Maybe we should lowercase the I in identity, as it might be confusing with the old `Identity` in KILT. Else, we mention DIDs as use those as the identity.

File: docs/getting-started.md [hunk 25]
```diff
@@ -80,99 +84,285 @@ await Kilt.init({ address: YOUR_CHAIN_ADDRESS })
...
+  seed: generateClaimerMnemonic,
 })
 ```
 
-### 3.2. Storing a CTYPE
+### 2.2. Generate a light DID for the Claimer
 
-Before you can store the CTYPE on the blockchain, you have to configure your blockchain address and connect to it.
-Either explicitly:
+Using the keys from the demo keystore to generate the claimers light DID.
```
Comment (Diiaablo95) [inline docs/getting-started.md:185]:
```suggest
Using the keys from the demo keystore to generate the claimer's light DID.
```

File: docs/getting-started.md [hunk 26]
```diff
@@ -80,99 +84,285 @@ await Kilt.init({ address: YOUR_CHAIN_ADDRESS })
...
-      type: 'integer',
-    },
-  },
-  type: 'object',
+const keystore = new Kilt.Did.DemoKeystore()
+
+// Signing keypair
+const claimerSigningKeypair = await keystore.generateKeypair({
+  alg: Kilt.Did.SigningAlgorithms.Ed25519,
+  seed: generateClaimerMnemonic,
```
Comment (Diiaablo95) [inline docs/getting-started.md:173]: Should this only be `claimerMnemonic`?

File: docs/getting-started.md [hunk 27]
```diff
@@ -80,99 +84,285 @@ await Kilt.init({ address: YOUR_CHAIN_ADDRESS })
...
+})
+
+// Encryption keypair
+const claimerEncryptionKeypair = await keystore.generateKeypair({
+  alg: Kilt.Did.EncryptionAlgorithms.NaclBox,
+  seed: generateClaimerMnemonic,
```
Comment (Diiaablo95) [inline docs/getting-started.md:179]: Should this only be `claimerMnemonic`?

File: docs/getting-started.md [hunk 28]
```diff
@@ -80,99 +84,285 @@ await Kilt.init({ address: YOUR_CHAIN_ADDRESS })
...
+  },
+  encryptionKey: {
+    publicKey: claimerEncryptionKeypair.publicKey,
+    type: Kilt.Did.DemoKeystore.getKeypairTypeForAlg(
+      claimerEncryptionKeypair.alg
+    ),
+  },
+})
+
+// DID light: `did:kilt:light:014qFxmHnWw5sGMwjskdvMCrASF9Jvu5ggWRTWTK2NNYSLDg56:oWFlomlwdWJsaWNLZXlYIJuIow7rjSdf92qMKYtWV42lF9mctD1nFf8RM24auJhwZHR5cGVmeDI1NTE5`
```
Comment (Diiaablo95) [inline docs/getting-started.md:203]: This won't probably match the output DID as we use a random mnemonic for the claimer.

File: docs/getting-started.md [hunk 29]
```diff
@@ -80,99 +84,285 @@ await Kilt.init({ address: YOUR_CHAIN_ADDRESS })
...
+console.log(claimerLightDid.did)
 ```
 
-Or by setting a default address in the configuration, connecting implicitly.
+### 2.3 Generate a full DID for the Attester
+
+Before we can send the request for an attestation to an Attester, we should first create an Attester identity and a full on-chain DID. Using the previously generated keyring.
```
Comment (Diiaablo95) [inline docs/getting-started.md:209]:
```suggest
identity and a full on-chain DID using the previously generated keyring.
```

File: docs/getting-started.md [hunk 30]
```diff
@@ -80,99 +84,285 @@ await Kilt.init({ address: YOUR_CHAIN_ADDRESS })
...
+await Kilt.Balance.makeTransfer(attester.address, transferAmount).then((tx) =>
+  Kilt.BlockchainUtils.signAndSubmitTx(tx, identity, {
+    resolveOn: Kilt.BlockchainUtils.IS_FINALIZED,
+    reSign: true,
+  })
+)
 ```
 
-initializes the SDK _and_ sets the config, so it is related to the second approach.
+Then we generate all the keypairs for the attester to construct the attestation, delegation and authentication keys with the keystore.
```
Comment (Diiaablo95) [inline docs/getting-started.md:237]: We should specify that we use the same key for authentication, attestation, and delegation but that this is not recommended for production use.

File: docs/getting-started.md [hunk 31]
```diff
@@ -80,99 +84,285 @@ await Kilt.init({ address: YOUR_CHAIN_ADDRESS })
...
+  reSign: true,
+  resolveOn: Kilt.BlockchainUtils.IS_FINALIZED,
+})
+
+// The resolved full DID
+const fullDid = await Did.DefaultResolver.resolveDoc(did)
+
+console.log('Full DID', fullDid.details)
 
+const fullDid = {
```
Comment (Diiaablo95) [inline docs/getting-started.md:300]: What is this?

File: docs/getting-started.md [hunk 32]
```diff
@@ -80,99 +84,285 @@ await Kilt.init({ address: YOUR_CHAIN_ADDRESS })
...
 ```typescript
-// the account behind this mnemonic already has tokens
-const identity = Kilt.Identity.buildFromMnemonic(
-  'receive clutch item involve chaos clutch furnace arrest claw isolate okay together',
-  // using ed25519 as key type because this is how the endowed identity is set up
-  { signingKeyPairType: 'ed25519' }
-)
 const tx = await ctype.store()
 
+const authorizedExtrinsic = await attesterOnChainDid.authorizeExtrinsic(
```
Comment (Diiaablo95) [inline docs/getting-started.md:356]: Where is `attesterOnChainDid` defined?

File: docs/getting-started.md [hunk 33]
```diff
@@ -181,8 +371,8 @@ await Kilt.BlockchainUtils.signAndSubmitTx(tx, identity, {
 
 // or step by step
 const chain = Kilt.connect()
```
Comment (Diiaablo95) [inline docs/getting-started.md:373]:
```suggest
const chain = await Kilt.connect()
```

File: docs/getting-started.md [hunk 34]
```diff
@@ -181,8 +371,8 @@ await Kilt.BlockchainUtils.signAndSubmitTx(tx, identity, {
...
-const signed = chain.signTx(identity, tx)
-await Kilt.BlockchainUtils.submitSignedTx(tx)
+const signed = chain.signTx(attester, authorizedExtrinsic)
```
Comment (Diiaablo95) [inline docs/getting-started.md:374]: Is this also a promise? Not sure.

File: docs/getting-started.md [hunk 35]
```diff
@@ -336,16 +523,34 @@ Message {
 }
 ```
 
-The message can be encrypted as follows:
+After the message has been created the key agreement of both the claimer and attester must be retreived in order for encryption. Using the previously generated DIDs of the claimer and attesters.
```
Comment (Diiaablo95) [inline docs/getting-started.md:526]:
```suggest
After the message has been created the key agreement of both the claimer and attester must be retrieved in order to encrypt each other's messages using the previously generated DIDs of the claimer and attesters.
```

File: docs/getting-started.md [hunk 36]
```diff
@@ -336,16 +523,34 @@ Message {
...
 
 ```typescript
-const encrypted = message.encrypt(claime, attester.getPublicIdentity())
+const claimerKeyAgreement = claimerLightDid.getKeys(
```
Comment (Diiaablo95) [inline docs/getting-started.md:529]:
```suggest
const claimerKeyAgreementKey = claimerLightDid.getKeys(
```

File: docs/getting-started.md [hunk 37]
```diff
@@ -336,16 +523,34 @@ Message {
...
+  KeyRelationship.keyAgreement
+)[0] as IDidKeyDetails<string>
+const attesterKeyAgreement = attesterOnChainDid.getKeys(
```
Comment (Diiaablo95) [inline docs/getting-started.md:532]:
```suggest
const attesterKeyAgreementKey = attesterOnChainDid.getKeys(
```

**Review by Diiaablo95 — CHANGES_REQUESTED** [review]
There are some inconsistencies especially in the `getting-started.md` file. I would suggest to go over the file once again from start to end, also considering the suggestions, and make any furter changes as needed. The same applies for the `getting-started.ts` file, which I have quickly gone over and seen some of the same "issue" I identified in the `getting-started.md`, probably because code duplication is exactly what we have done here.

File: docs/example.ts (hunk 1 above)
Comment (Dudleyneedham) [reply docs/example.ts:49]: No, since it is running off the CI pipeline which requires the faucet to be `ed25519` for the standalone verison.

File: docs/example.ts (hunk 2 above)
Comment (Dudleyneedham) [reply docs/example.ts:60]: I wanted to keep it consistent throughout.

File: docs/example.ts (hunk 3 above)
Comment (Dudleyneedham) [reply docs/example.ts:112]: Agreed.

File: docs/example.ts (hunk 4 above)
Comment (Dudleyneedham) [reply docs/example.ts:134]: yes, this is left over.

File: docs/example.ts (hunk 5 above)
Comment (Dudleyneedham) [reply docs/example.ts:138]: Ok, sounds better. I have added the following:

`// Generate authentication and encryption keys used to derive a light DID from them.`

File: docs/example.ts (hunk 6 above)
Comment (Dudleyneedham) [reply docs/example.ts:147]: `// Using the generated authentication and encryption keys to derive a light DID.`

File: docs/example.ts (hunk 7 above)
Comment (Dudleyneedham) [reply docs/example.ts:163]: Updated.

File: docs/example.ts (hunk 8 above)
Comment (Dudleyneedham) [reply docs/example.ts:165]: Changed to account.

File: docs/example.ts (hunk 9 above)
Comment (Dudleyneedham) [reply docs/example.ts:222]: Updated.

File: docs/example.ts (hunk 10 above)
Comment (Dudleyneedham) [reply docs/example.ts:225]: Updated.

File: docs/example.ts (hunk 11 above)
Comment (Dudleyneedham) [reply docs/example.ts:257]: Good catch.

File: docs/example.ts (hunk 12 above)
Comment (Dudleyneedham) [reply docs/example.ts:337]: Updated.

File: docs/example.ts (hunk 13 above)
Comment (Dudleyneedham) [reply docs/example.ts:353]: Updated.

File: docs/example.ts (hunk 14 above)
Comment (Dudleyneedham) [reply docs/example.ts:356]: Updated.

File: docs/example.ts (hunk 15 above)
Comment (Dudleyneedham) [reply docs/example.ts:422]: Updated.

File: docs/example.ts (hunk 16 above)
Comment (Dudleyneedham) [reply docs/example.ts:425]: updated.

File: docs/getting-started.md (hunk 17 above)
Comment (Dudleyneedham) [reply docs/getting-started.md:15]: I have updated all the places and removed the `how to`

File: docs/getting-started.md (hunk 18 above)
Comment (Dudleyneedham) [reply docs/getting-started.md:105]: Yes, it should have the await.

File: docs/getting-started.md (hunk 19 above)
Comment (Dudleyneedham) [reply docs/getting-started.md:113]: After our discussion, I have renamed it to `test-net`.

File: docs/getting-started.md (hunk 20 above)
Comment (Dudleyneedham) [reply docs/getting-started.md:117]: Updated.

File: docs/getting-started.md (hunk 22 above)
Comment (Dudleyneedham) [reply docs/getting-started.md:120]: Updated.

File: docs/getting-started.md (hunk 23 above)
Comment (Dudleyneedham) [reply docs/getting-started.md:123]: Updated.

File: docs/getting-started.md (hunk 24 above)
Comment (Dudleyneedham) [reply docs/getting-started.md:139]: Yes, that is a good point.

Comment (Dudleyneedham) [reply docs/getting-started.md:139]: I substituted the identity for an account.

Comment (Dudleyneedham) [reply docs/getting-started.md:139]: Let me know what you think.

File: docs/getting-started.md (hunk 25 above)
Comment (Dudleyneedham) [reply docs/getting-started.md:185]: Sounds good.

File: docs/getting-started.md (hunk 26 above)
Comment (Dudleyneedham) [reply docs/getting-started.md:173]: Yes, I will update this.

File: docs/getting-started.md (hunk 27 above)
Comment (Dudleyneedham) [reply docs/getting-started.md:179]: Updated.

File: docs/getting-started.md (hunk 28 above)
Comment (Dudleyneedham) [reply docs/getting-started.md:203]: Sure, I will put `Example light DID:`.

File: docs/getting-started.md (hunk 29 above)
Comment (Dudleyneedham) [reply docs/getting-started.md:209]: Sure, sounds good.

File: docs/getting-started.md (hunk 30 above)
Comment (Dudleyneedham) [reply docs/getting-started.md:237]: I have added a note.

File: docs/getting-started.md (hunk 31 above)
Comment (Dudleyneedham) [reply docs/getting-started.md:300]: I have added it as an example.

File: docs/getting-started.md (hunk 32 above)
Comment (Dudleyneedham) [reply docs/getting-started.md:356]: I have changed `fullDid` to be `attesterOnChainDid`

File: docs/getting-started.md (hunk 33 above)
Comment (Dudleyneedham) [reply docs/getting-started.md:373]: Updated.

File: docs/getting-started.md (hunk 34 above)
Comment (Dudleyneedham) [reply docs/getting-started.md:374]: Yes, I have updated.

File: docs/getting-started.md (hunk 35 above)
Comment (Dudleyneedham) [reply docs/getting-started.md:526]: Updated.

File: docs/getting-started.md (hunk 37 above)
Comment (Dudleyneedham) [reply docs/getting-started.md:532]: Updated.

File: docs/example.ts (hunk 14 above)
Comment (Diiaablo95) [reply docs/example.ts:354]:
```suggest
  const verifierEncryptionKey = verifierLightDid.getKeys(
```

File: docs/getting-started.ts [hunk 38]
```diff
@@ -36,19 +76,32 @@ async function main(): Promise<void> {
...
   /* To store the CTYPE on the blockchain, you have to call: */
-  const identity = Kilt.Identity.buildFromMnemonic(
-    'receive clutch item involve chaos clutch furnace arrest claw isolate okay together',
-    // using ed25519 as key type because this is how the endowed identity is set up
-    { signingKeyPairType: 'ed25519' }
+  const accountMnemonic =
+    'receive clutch item involve chaos clutch furnace arrest claw isolate okay together'
+  const account = keyring.addFromMnemonic(accountMnemonic)
+
+  const onChainDid = await Kilt.Did.createOnChainDidFromSeed(
```
Comment (Diiaablo95) [inline docs/getting-started.ts:84]: I would perhaps rename it to `fullDid`? I know the method says `createOnChainDidFromSeed` but using a different name might be confusing.

File: docs/getting-started.ts [hunk 39]
```diff
@@ -194,28 +281,53 @@ async function main(): Promise<void> {
...
-        const isValid = await Kilt.AttestedClaim.fromAttestedClaim(
+        const claims = decryptedMessageForVerifier.body.content
+        console.log('before verifying', credentialForVerifier)
+
+        const verifiablePresentation = Kilt.AttestedClaim.fromAttestedClaim(
           claims[0]
-        ).verify()
+        )
+        const isValid = await verifiablePresentation.verify()
         console.log('Verifcation success?', isValid)
```
Comment (Diiaablo95) [inline docs/getting-started.ts:327]:
```suggest
        console.log('Verification success?', isValid)
```

File: docs/getting-started.md (hunk 26 above)
Comment (Diiaablo95) [reply docs/getting-started.md:175]: Has this been updated?

File: docs/getting-started.md (hunk 27 above)
Comment (Diiaablo95) [reply docs/getting-started.md:181]: I don't see the change?

File: docs/getting-started.md (hunk 32 above)
Comment (Diiaablo95) [reply docs/getting-started.md:359]: As I said, perhaps `onChainDid` is more confusing than `fullDid`.

**Review by Diiaablo95 — COMMENTED** [review]
Some minor changes and then, at least for me, we're good to go!

File: docs/getting-started.ts (hunk 39 above)
Comment (Dudleyneedham) [reply docs/getting-started.ts:327]: updated

**Review by Diiaablo95 — APPROVED** [review]
✅ ✅ ✅ ✅ ✅ ✅ ✅ ✅
